{"cells":[{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":21744,"status":"ok","timestamp":1675763304752,"user":{"displayName":"Alan Nguyen","userId":"13588881381918321795"},"user_tz":-480},"id":"bmXxOB6Oo2jL","outputId":"5d791fdf-7b6d-4824-d7c0-34555d075d05"},"outputs":[{"output_type":"stream","name":"stdout","text":["Mounted at /content/drive\n"]}],"source":["from google.colab import drive\n","drive.mount('/content/drive')"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"_lbX__ZMngts","outputId":"6f36c7d3-4e6d-4fb3-dbcf-2a12f58d4404","executionInfo":{"status":"ok","timestamp":1675765465728,"user_tz":-480,"elapsed":58286,"user":{"displayName":"Alan Nguyen","userId":"13588881381918321795"}}},"outputs":[{"output_type":"stream","name":"stdout","text":["Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n","Requirement already satisfied: torpy in /usr/local/lib/python3.8/dist-packages (1.1.6)\n","Requirement already satisfied: cryptography>=3.2 in /usr/local/lib/python3.8/dist-packages (from torpy) (39.0.0)\n","Requirement already satisfied: cffi>=1.12 in /usr/local/lib/python3.8/dist-packages (from cryptography>=3.2->torpy) (1.15.1)\n","Requirement already satisfied: pycparser in /usr/local/lib/python3.8/dist-packages (from cffi>=1.12->cryptography>=3.2->torpy) (2.21)\n"]},{"output_type":"stream","name":"stderr","text":["ERROR:torpy.cell_socket:timed out\n","ERROR:root:[ignored]\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/cell_socket.py\", line 63, in connect\n","    self._socket.connect((self._router.ip, self._router.or_port))\n","  File \"/usr/lib/python3.8/ssl.py\", line 1342, in connect\n","    self._real_connect(addr, False)\n","  File \"/usr/lib/python3.8/ssl.py\", line 1329, in _real_connect\n","    super().connect(addr)\n","socket.timeout: timed out\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/utils.py\", line 79, in newfn\n","    return func(*args, **kwargs)\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/consesus.py\", line 235, in renew\n","    raw_string = self.download_consensus(prev_hash)\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/consesus.py\", line 183, in newfn\n","    return func(*args, **kwargs)\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/consesus.py\", line 390, in download_consensus\n","    with self._get_dir_client() as dir_client:\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/consesus.py\", line 375, in _get_dir_client\n","    self._dir_guard, self._dir_circuit = self._create_dir_circuit(purpose='Internal dir client')\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/consesus.py\", line 365, in _create_dir_circuit\n","    guard = TorGuard(router, purpose=purpose)\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/guard.py\", line 66, in __init__\n","    self.__tor_socket.connect()\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/cell_socket.py\", line 69, in connect\n","    raise TorSocketConnectError(e)\n","torpy.cell_socket.TorSocketConnectError: timed out\n","WARNING:torpy.utils:Retry with another router...\n","ERROR:torpy.cell_socket:[Errno 113] No route to host\n","ERROR:root:[ignored]\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/cell_socket.py\", line 63, in connect\n","    self._socket.connect((self._router.ip, self._router.or_port))\n","  File \"/usr/lib/python3.8/ssl.py\", line 1342, in connect\n","    self._real_connect(addr, False)\n","  File \"/usr/lib/python3.8/ssl.py\", line 1329, in _real_connect\n","    super().connect(addr)\n","OSError: [Errno 113] No route to host\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/utils.py\", line 79, in newfn\n","    return func(*args, **kwargs)\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/consesus.py\", line 235, in renew\n","    raw_string = self.download_consensus(prev_hash)\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/consesus.py\", line 183, in newfn\n","    return func(*args, **kwargs)\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/consesus.py\", line 390, in download_consensus\n","    with self._get_dir_client() as dir_client:\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/consesus.py\", line 375, in _get_dir_client\n","    self._dir_guard, self._dir_circuit = self._create_dir_circuit(purpose='Internal dir client')\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/consesus.py\", line 365, in _create_dir_circuit\n","    guard = TorGuard(router, purpose=purpose)\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/guard.py\", line 66, in __init__\n","    self.__tor_socket.connect()\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/cell_socket.py\", line 69, in connect\n","    raise TorSocketConnectError(e)\n","torpy.cell_socket.TorSocketConnectError: [Errno 113] No route to host\n","WARNING:torpy.utils:Retry with another router...\n","ERROR:torpy.cell_socket:timed out\n","ERROR:root:[ignored]\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/cell_socket.py\", line 63, in connect\n","    self._socket.connect((self._router.ip, self._router.or_port))\n","  File \"/usr/lib/python3.8/ssl.py\", line 1342, in connect\n","    self._real_connect(addr, False)\n","  File \"/usr/lib/python3.8/ssl.py\", line 1329, in _real_connect\n","    super().connect(addr)\n","socket.timeout: timed out\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/utils.py\", line 79, in newfn\n","    return func(*args, **kwargs)\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/consesus.py\", line 235, in renew\n","    raw_string = self.download_consensus(prev_hash)\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/consesus.py\", line 183, in newfn\n","    return func(*args, **kwargs)\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/consesus.py\", line 390, in download_consensus\n","    with self._get_dir_client() as dir_client:\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/consesus.py\", line 375, in _get_dir_client\n","    self._dir_guard, self._dir_circuit = self._create_dir_circuit(purpose='Internal dir client')\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/consesus.py\", line 365, in _create_dir_circuit\n","    guard = TorGuard(router, purpose=purpose)\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/guard.py\", line 66, in __init__\n","    self.__tor_socket.connect()\n","  File \"/usr/local/lib/python3.8/dist-packages/torpy/cell_socket.py\", line 69, in connect\n","    raise TorSocketConnectError(e)\n","torpy.cell_socket.TorSocketConnectError: timed out\n","WARNING:torpy.utils:Retry with another router...\n","WARNING:torpy.consesus:Unknown voter present\n","WARNING:torpy.consesus:Unknown voter present\n"]},{"output_type":"stream","name":"stdout","text":["{'origin': '109.70.100.24'}\n","{'code': 0, 'msg': \"Service unavailable from a restricted location according to 'b. Eligibility' in https://www.binance.com/en/terms. Please contact customer service if you believe you received this message in error.\"}\n"]}],"source":["#remark: \"mount Drive\" before running the code\n","#put this python notebook file under the src directory\n","\n","#to refresh the code properly: should pick \"Restart and run all\", \n","#otherwise: \"Runtime --> disconnect and delete runtime\"\n","\n","import warnings\n","warnings.filterwarnings('ignore')\n","\n","import os\n","import sys\n","import json\n","import requests\n","\n","#hyper-parameters\n","project_path = \"drive/MyDrive/investment/tung/tung/\"\n","sys.path.insert(0, project_path)\n","import constants\n","\n","BINANE_URL = 'https://api.binance.com/api/v1/'\n","exchange_url = BINANE_URL + 'exchangeInfo'\n","\n","!pip install torpy\n","from torpy.http.requests import TorRequests\n","\n","with TorRequests() as tor_requests:\n","  with tor_requests.get_session() as sess:\n","    print(sess.get(\"http://httpbin.org/ip\").json())\n","\n","\n","    data = json.loads(requests.get(exchange_url).text)\n","    print (data)\n","\n","# import crypto_bot\n","# crypto_bot.run_bot(0)\n"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"SvbuprO-IAE3"},"outputs":[],"source":[]}],"metadata":{"colab":{"provenance":[{"file_id":"1hNYXXG_fWecDuXUWgbB86flLghZ245Sz","timestamp":1666800091753},{"file_id":"1hmOrSM8itBjLltmrstYBsTqF_enbO2Ek","timestamp":1662017834882}]},"gpuClass":"standard","kernelspec":{"display_name":"Python 3.10.6 64-bit","language":"python","name":"python3"},"language_info":{"name":"python","version":"3.10.6"},"vscode":{"interpreter":{"hash":"b0fa6594d8f4cbf19f97940f81e996739fb7646882a419484c72d19e05852a7e"}}},"nbformat":4,"nbformat_minor":0}